legend-engine-xt-nonrelationalStore-mongodb-executionPlan-test/src/main/resources/core_mongodb_execution_test/mongodb_execution_tests.pure

```diff
@@ -0,0 +1,488 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import meta::json::*;
+import meta::pure::graphFetch::execution::*;
+import meta::pure::mapping::*;
+import meta::pure::extension::*;
+import meta::pure::functions::io::http::*;
+import meta::pure::test::*;
+import meta::pure::store::*;
+import meta::pure::runtime::*;
+import meta::external::store::mongodb::executionTest::utils::*;
+import meta::external::store::mongodb::executionPlan::legendJava::test::*;
+
+import meta::external::store::mongodb::executionTest::*;
+import meta::external::store::mongodb::executionTest::utils::*;
+import meta::external::store::mongodb::executionTest::testCase::utils::*;
+import meta::external::store::mongodb::executionTest::domain::*;
+
+Class meta::external::store::mongodb::executionTest::TestConfig
+{
+  tag: String[1];
+  mapping: meta::pure::mapping::Mapping[1];
+  runtime: meta::pure::runtime::Runtime[1];
+}
+
+function meta::external::store::mongodb::executionTest::getCompiledTestConfig(tag: String[1], mongoDBPort: Integer[1], pureSetupToCompileFileLoc: String[1]):meta::external::store::mongodb::executionTest::TestConfig[1]
+{
+
+let tokenizedFile = meta::pure::functions::io::readFile($pureSetupToCompileFileLoc)->toOne();
+
+// Is there a better way to both compile our DSL grammar and inject the randomized mongodb port?
+let detokenizedFile = $tokenizedFile->replace('serverURLs: [localhost:__PORT_TOKEN__];', 'serverURLs: [localhost:' + $mongoDBPort->toString() + '];');
+
+let compiled = meta::legend::compile($detokenizedFile);
+let compiledMapping = $compiled->at(3)->cast(@meta::pure::mapping::Mapping)->toOne();
+let compiledRuntime = $compiled->at(1)->cast(@meta::pure::runtime::PackageableRuntime).runtimeValue->cast(@meta::pure::runtime::Runtime);
+
+ let testConfig = ^meta::external::store::mongodb::executionTest::TestConfig(
+        tag = $tag,
+        mapping = $compiledMapping,
+        runtime = $compiledRuntime
+      );
+}
+
+function  meta::external::store::mongodb::executionTest::collectTest(
+    tag: String[1],
+    testConfig: TestConfig[1]
+  ): PureTestCollection[1]
+{
+ println('meta::external::store::mongodb::executionTest::collectTest');
+  meta::pure::test::collectParameterizedTests(
+    meta::external::store::mongodb::executionTest::testCase,
+    $tag,
+    $testConfig,
+    [],
+    []
+  );
+}
+
+function <<paramTest.BeforePackage>> meta::external::store::mongodb::executionTest::testCase::startMongoDB(config:TestConfig[1]):Boolean[1]
+{
+    println(' meta::external::store::mongodb::executionTest::testCase::startMongoDB');
```

Comment:
we don't use this? right

Reply:
good catch, we do this instead in the collect tests portion. removed.